docs/provider/pulumi.md

```diff
@@ -0,0 +1,51 @@
+## Pulumi ESC
+
+Sync environments, configs and secrets from [Pulumi ESC](https://www.pulumi.com/product/esc/) to Kubernetes using the External Secrets Operator.
+
+### Authentication
+
+Pulumi [Access Tokens](https://www.pulumi.com/docs/pulumi-cloud/access-management/access-tokens/) are recommended to access Pulumi ESC.
+
+### Creating a SecretStore
+
+A Pulumi SecretStore can be created by specifying the `organization` and `environment` and referencing a Kubernetes secret containing the `accessToken`.
+
+```yaml
+apiVersion: external-secrets.io/v1beta1
+kind: SecretStore
+metadata:
+  name: secret-store
+spec:
+  provider:
+    pulumi:
+      organization: <NAME_OF_THE_ORGANIZATION>
+      environment: <NAME_OF_THE_ENVIRONMENT>
+      accessToken:
+        secretRef:
+          name: <NAME_OF_KUBE_SECRET>
+          key: <KEY_IN_KUBE_SECRET>
+```
+
+If required, the API URL (`apiUrl`) can be customized as well. If not specified, the default value is `https://api.pulumi.com`.
+
+Additionally, the `duration` can be customized, which specifies how long the session is valid. If not specified, the default value is `5m`.
+
+### Referencing Secrets
+
+Secrets can be referenced by defining the `key` containing the JSON path to the secret. Pulumi ESC secrets are internally organized as a JSON object.
+
+```yaml
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: secret
+spec:
+  refreshInterval: 20s
+  secretStoreRef:
+    kind: SecretStore
+    name: secret-store
+  data:
+  - secretKey: <KEY_IN_KUBE_SECRET>
+    remoteRef:
+      key: <SECRET_JSON_PATH>
```

Comment:
I think it would be great to provide details and an example here. E.g. you can use `foo.bar` to descend a tree, use `foo.baz[3]` to index an array etc. 
Is this following the [JSONPath spec](https://www.ietf.org/archive/id/draft-goessner-dispatch-jsonpath-00.html) or is it something similar or completely different?

Are there upstream API docs available?

Reply:
Done, added some examples for the Pulumi path syntax and link to the docs. In the future there will be a dedicated page for the Pulumi syntax. I talked with the engineers about it.